src/pages/reset-password/[resettoken].tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useRouter } from 'next/router';
+import { useApiClient } from '@common/contexts';
+
+import { HStack, VStack } from '@chakra-ui/react';
+import { AppContainer, Logo } from '@src/common/components';
+import { ResetPasswordForm } from '@src/modules/auth';
+import { AuthRoute } from '@src/constants/PageRoutes';
+import { useResetPasswordMutation } from '@generated/graphql.queries';
+import { isNil } from 'lodash';
+
+export default function ForgetPassword() {
+  const { t } = useTranslation('auth');
+  const { gqlClient } = useApiClient();
+  const head = { title: t('reset-password.title') };
+
+  const [resetToken, setResetToken] = useState('');
+  const [successMessage, setSuccessMessage] = useState('');
+  const [successAction, setSuccessAction] = useState('');
+  const [successActionLink, setSuccessActionLink] = useState('');
+
+  const token = useRouter().query.resettoken as string;
+  useEffect(() => {
+    !isNil(token) && setResetToken(token);
+  }, [token]);
+
+  const {
+    mutate: resetPassword,
+    error,
+    isLoading: isResetingEmail,
+  } = useResetPasswordMutation(gqlClient, {
+    onSettled: (data, error) => {
+      if (error) {
+        console.error(error);
+        setSuccessMessage('');
+      }
+      if (data) {
+        setSuccessMessage(t('reset-password.success'));
+        setSuccessAction(t('reset-password.success-action'));
+        setSuccessActionLink(AuthRoute.Login);
+      }
+    },
+  });
+
+  const onSubmit = (value: ResetPasswordFormTypes) => {
+    resetPassword({
+      detail: {
+        password: value.password,
+        resetToken,
+      },
+    });
+  };
+
+  return (
+    <AppContainer head={head}>
+      <VStack flex={1} padding="1rem">
+        <HStack width="100%">
+          <Logo height="5rem" />
+        </HStack>
+        <ResetPasswordForm
+          onSubmit={onSubmit}
+          isLoading={isResetingEmail}
+          isInvalid={!isNil(error)}
+          successMessage={successMessage}
```

Comment:
Instead of letting the form handle the action after the mutation succeeded, which it shouldn't really.

`react-query` [useMutation](https://react-query.tanstack.com/reference/useMutation) has this `status` value to let us know whether this mutation has been called or not.

By leveraging this field, we can have following pattern set up.

```
{ status === 'idle' && <ResetPasswordForm/> }
{ status === 'success' && <PasswordResetNotification/> } // You don't have to make this a component btw.
```

PS. since graphql endpoint will always return 200 if a response came back, so you need to find a different solution to check `status === 'success'`

Reply:
I updated this one by redirecting users to login page when success